test/src/test/java/org/corfudb/integration/CorfuReplicationLargeTxIT.java

```diff
@@ -0,0 +1,324 @@
+            LogReplicationMetadata.ReplicationStatusKey.class,
+            LogReplicationMetadata.ReplicationStatusVal.class,
+            null,
+            TableOptions.fromProtoSchema(LogReplicationMetadata.ReplicationStatusVal.class));
+
+        CountDownLatch statusUpdateLatch = new CountDownLatch(totalStandbyStatusUpdates);
+        ReplicationStatusListener standbyListener =
+            new ReplicationStatusListener(statusUpdateLatch, false);
+        corfuStoreStandby.subscribeListener(standbyListener, LogReplicationMetadataManager.NAMESPACE,
+            LogReplicationMetadataManager.LR_STATUS_STREAM_TAG);
+
+        // Calculate the expected total number of streaming updates across all
+        // tables
+        int totalStreamingUpdates =
+            expectedStreamingUpdatesPerTable * mapNameToMapStandby.size();
+
+        CountDownLatch streamingUpdatesLatch =
+            new CountDownLatch(totalStreamingUpdates);
+        StreamingUpdateListener streamingUpdateListener =
+            new StreamingUpdateListener(streamingUpdatesLatch);
+        corfuStoreStandby.subscribeListener(streamingUpdateListener,
+            DefaultLogReplicationConfigAdapter.NAMESPACE,
+            DefaultLogReplicationConfigAdapter.TAG_ONE);
+
+        startLogReplicatorServersWithCustomMaxWriteSize(maxWriteSize);
+
+        log.debug("Wait for snapshot sync to finish");
+        statusUpdateLatch.await();
+        streamingUpdatesLatch.await();
+
+        // Verify that updates were received for all replicated tables with data
+        Assert.assertEquals(mapNameToMapStandby.size(),
+            streamingUpdateListener.getTableNameToUpdatesMap().size());
+
+        mapNameToMapStandby.keySet().forEach(key ->
+            Assert.assertTrue(streamingUpdateListener.getTableNameToUpdatesMap()
+                .containsKey(key)));
+
+
+        // Verify that the right number of entries are contained in the
+        // streaming update.  Also verify that the first entry is a 'clear'
+        // followed by all updates.
+        for (List<List<CorfuStreamEntry<Sample.StringKey,
+            SampleSchema.ValueFieldTagOne, Sample.Metadata>>> outerList :
+            streamingUpdateListener.getTableNameToUpdatesMap().values()) {
+
+            Assert.assertEquals(expectedStreamingUpdatesPerTable, outerList.size());
+
+            int totalEntriesReceived = 0;
+            List<CorfuStreamEntry<Sample.StringKey,
+                SampleSchema.ValueFieldTagOne, Sample.Metadata>> updateEntries =
+                new ArrayList<>();
+
+            for (int i=0; i<outerList.size(); i++) {
+                // If the snapshot apply was chunked, the first entry in the
+                // first chunk must be a 'clear'
+                if (i == 0) {
+                    Assert.assertEquals(CorfuStreamEntry.OperationType.CLEAR,
+                        outerList.get(i).get(0).getOperation());
+                    for (int j = 1; j<outerList.get(i).size(); j++) {
```

Comment:
A simpler logic would be to add a slice by using: list.subList() (you add everything from index 1 to outerList.get(i).size()) and no need for another for loop...